packages/client/src/runtime/utils/decimalJsLike.ts

```diff
@@ -25,12 +25,12 @@ export function isDecimalJsLike(value: unknown): value is DecimalJsLike {
 
 export function stringifyDecimalJsLike(value: DecimalJsLike): string {
   if (Decimal.isDecimal(value)) {
-    return String(value)
+    return JSON.stringify(String(value))
   }
 
-  const tmpDecimal = new Decimal(0) as DecimalJsLike
-  tmpDecimal.d = value.d
-  tmpDecimal.e = value.e
-  tmpDecimal.s = value.s
-  return String(tmpDecimal)
+  const tmpDecimal = new Decimal(0)
+  ;(tmpDecimal as DecimalJsLike).d = value.d
```

Comment:
What is the reason for this change?

Reply:
Previously, cast was done right after tmp instance creation. This is needed because `d`, `p`, `q` are not writable properties.
However, `DecimalJSLike` has no `toFixed` method defined and adding it to the signature now theoretically would be a breaking change. So instead, we keep `tmpDecimal` type to `Decimal` and cast to `DecimalJSLike` when we need writable property.

Reply:
Thanks for the thorough explanation 👍